packages/react-textarea/src/components/TextArea/useTextAreaStyles.ts

```diff
@@ -0,0 +1,246 @@
+  //   },
+  //   ':focus-within:after': {
+  //     // Animation for focus IN
+  //     transform: 'scaleX(1)',
+  //     transitionProperty: 'transform',
+  //     transitionDuration: motionDurations.normal,
+  //     transitionDelay: motionCurves.decelerateMid,
+  //   },
+  //   ':focus-within:active:after': {
+  //     // This is if the user clicks the field again while it's already focused
+  //     borderBottomColor: tokens.colorCompoundBrandStrokePressed,
+  //   },
+  //   ':focus-within': {
+  //     outlineWidth: tokens.strokeWidthThick,
+  //     outlineStyle: 'solid',
+  //     outlineColor: 'transparent',
+  //   },
+  // },
+
+  filledDarker: {
+    backgroundColor: tokens.colorNeutralBackground3,
+    ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorTransparentStrokeInteractive),
+  },
+
+  filledLighter: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorTransparentStrokeInteractive),
+  },
+
+  outline: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStroke1),
+    borderBottomColor: tokens.colorNeutralStrokeAccessible,
+  },
+  outlineInteractive: {
+    ':hover': {
+      ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStroke1Hover),
+      borderBottomColor: tokens.colorNeutralStrokeAccessibleHover,
+    },
+
+    ':active': {
+      ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStroke1Pressed),
+      borderBottomColor: tokens.colorNeutralStrokeAccessiblePressed,
+    },
+
+    ':focus-within': {
+      ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStroke1),
+      borderBottomColor: tokens.colorCompoundBrandStroke,
+    },
+  },
+});
+
+/**
+ * Styles for the textArea slot
+ */
+const useTextAreaStyles = makeStyles({
+  base: {
+    ...shorthands.borderStyle('none'),
+    ...shorthands.margin('0'),
+    backgroundColor: 'transparent',
+    color: tokens.colorNeutralForeground1,
+
+    '::placeholder': {
+      color: tokens.colorNeutralForeground4,
+      opacity: 1,
+    },
+
+    '::selection': {
+      color: tokens.colorNeutralForegroundInverted,
+      backgroundColor: tokens.colorNeutralBackgroundInverted,
+    },
+
+    ':focus-visible': {
+      outlineStyle: 'none', // disable default browser outline
+    },
+  },
+
+  none: {
+    resize: 'none',
+  },
+  both: {
+    resize: 'both',
+  },
+  horizontal: {
+    resize: 'horizontal',
+  },
+  vertical: {
+    resize: 'vertical',
+  },
+
+  // The padding style adds both content and regular padding (from design spec), this is because the handle is not
+  // affected by changing the padding of the root.
+  small: {
+    height: textAreaHeight.small,
+    ...shorthands.padding(
+      '0',
+      `calc(${spacingTokens.horizontal.mNudge} + ${spacingTokens.horizontal.xss})`,
```

Comment:
Would it make sense to create a token like
```js
textAreaHorizontalPadding = {
 small: smallspacingTokens.horizontal.mNudge + spacingTokens.horizontal.xss`,
 medium: etc.,
 large: etc.
}
```
in this file? That way we don't need to browser to run `calc` for us.

Reply:
If the plan is for tokens to ultimately come from `react-theme` I think we'll need to use `calc` since those tokens resolve to CSS vars.

Reply:
Do you know if there's a plan any time soon to add the spacing tokens to the theme? At least to me it seems like they obviously should be there, but they're not for some reason.

Reply:
@miroslavstastny  for spacing tokens in theme question